### PR TITLE
Support ImageFilter.BuiltinFilter for I;16* images

### PR DESCRIPTION
**Pull request by radarhere**
Helps #6667

### CODE REVIEW

File: Tests/test_image_filter.py [hunk 1]
```diff
@@ -35,16 +35,25 @@
...
+)
+def test_sanity(
+    filter_to_apply: ImageFilter.Filter | type[ImageFilter.Filter], mode: str
+) -> None:
     im = hopper(mode)
-    if mode != "I" or isinstance(filter_to_apply, ImageFilter.BuiltinFilter):
+    if mode[0] != "I" or (
+        callable(filter_to_apply)
+        and issubclass(filter_to_apply, ImageFilter.BuiltinFilter)
+    ):
```
Comment (radarhere) [inline Tests/test_image_filter.py:48]: The `isinstance` condition, added in #7108, was never true - each `BuiltinFilter` run of this test uses a `BuiltinFilter` class, not an instance. So this corrects it.

File: src/libImaging/Filter.c [hunk 2]
```diff
@@ -26,6 +26,8 @@
 
 #include "Imaging.h"
 
+#define ROUND_UP(f) ((int)((f) >= 0.0 ? (f) + 0.5F : (f) - 0.5F))
```
Comment (hugovk) [inline src/libImaging/Filter.c:29]: Can we make this a function instead?

Comment (radarhere) [reply src/libImaging/Filter.c:29]: Any particular reason?

This was copying https://github.com/python-pillow/Pillow/blob/27c1bb265432cb5a1138e39165f0610e2d8a4e94/src/libImaging/Reduce.c#L5 and https://github.com/python-pillow/Pillow/blob/27c1bb265432cb5a1138e39165f0610e2d8a4e94/src/libImaging/Resample.c#L5

If there's a reason to prefer functions, then maybe they should all be changed.

I'd suggest a common function for all three, except that this uses a `float` and those other two use `double`.

Comment (hugovk) [reply src/libImaging/Filter.c:29]: Functions are generally safer than macros, which need extra care around parentheses and don't get extra checking. But if this is a copy/paste of existing ones then we should be fine.

**Review by hugovk — APPROVED** [review]

**Pull request merged**